src/main/java/software/amazon/awssdk/s3accessgrants/plugin/internal/S3AccessGrantsUtils.java

```diff
@@ -55,4 +62,63 @@ public static void argumentNotNull(Object param, String message) {
         }
     }
 
+    /**
+     * Gets the Lowest Common Ancestor (LCA) for all S3 resources defined.
+     * This allows us to combine credentials to access paths like "s3://a/b/c" and "s3://a/b/d"
+     * using a singular set of credentials (which may be required for calls like CopyObject or
+     * DeleteObjects). In this case, the LCA of the example would be "s3://a/b" and we would use
+     * that as our final resource to query S3AG.
+     * @param s3Resources List of S3Resources we would like to find the LCA for
+     * @return A singular S3 path that is the LCA of all the inputted resources.
+     */
+    public static String getLowestCommonAncestorPath(Collection<String> s3Resources) {
```

Comment:
You should perform per character backward string match. 
Eg: LCA of s3://bucket/folder/path1/logs and s3://bucket/folder/path1/logs should be s3://bucket/folder/path*